src/modules/RA/DataLayer/components/tabs/PopupTab/PopupFieldList.js

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback } from 'react';
+import Button from '@material-ui/core/Button';
+import DeleteIcon from '@material-ui/icons/Delete';
+import FormControl from '@material-ui/core/FormControl';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+
+import { makeStyles } from '@material-ui/core/styles';
+import { useForm } from 'react-final-form';
+
+const useStyles = makeStyles({
+  row: {
+    zIndex: 10,
+    margin: '1em 0',
+    backgroundColor: 'white',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+  },
+  formControl: {
+    display: 'flex',
+    flexDirection: 'columns',
+    alignItems: 'strech',
+  },
+});
+
+const PopupFieldRow = ({ popupField, onChange }) => {
```

Comment:
Can you use React.memo here to enhance perfs ?

Reply:
Done.